packages/components/src/components/input/input.stories.ts

```diff
@@ -0,0 +1,642 @@
+
+          const isValid = sdInput => sdInput.checkValidity();
+
+          if (sdInputs.every(isValid)) {
+            event.preventDefault(); // Prevent the default form submission behavior
+
+            const formData = new FormData(form);
+            const formValues = Object.fromEntries(formData);
+
+            alert('Form submitted successfully with the following values: ' + JSON.stringify(formValues, null, 2));
+          }
+        }
+
+        document.querySelector('#testForm').addEventListener('submit', handleSubmit);
+      </script>
+    `;
+  }
+};
+
+/**
+ * `sd-input` is fully accessibile via keyboard.
+ */
+
+export const Mouseless = {
+  render: (args: any) => {
+    return html`<div class="mouseless w-[231px]">${generateTemplate({ args })}</div>`;
+  },
+
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('.mouseless sd-input');
+    await waitUntil(() => el?.shadowRoot?.querySelector('input'));
+    el?.shadowRoot?.querySelector('input')!.focus();
+  }
+};
+
+/**
+ * Step function used in Stepper sample.
+ */
+const step = (direction: 'up' | 'down') => {
+  // Get input element and buttons
+  const inputEl = document.querySelector('sd-input')!;
+  const stepDownButton = document.querySelector('#stepDownButton')!;
+
+  // function to add class if not present
+  const addClass = (el: Element, className: string) => {
+    if (!el.classList.contains(className)) {
+      el.classList.add(className);
+    }
+  };
+
+  // function to remove class if present
+  const removeClass = (el: Element, className: string) => {
+    if (el.classList.contains(className)) {
+      el.classList.remove(className);
+    }
+  };
+
+  // function to adjust classlist of stepDownButton
+  const adjustClasslist = () => {
+    if (inputEl.value === '0') {
+      addClass(stepDownButton, 'text-neutral-500');
+    } else {
+      removeClass(stepDownButton, 'text-neutral-500');
+    }
+  };
+
+  // Call sd-input stepUp() or stepDown() method
+  if (direction === 'up') inputEl.stepUp();
+  if (direction === 'down') inputEl.stepDown();
+
+  // Adjust input value to 2 decimals (currency)
+  inputEl.value = String(parseInt(inputEl.value, 10).toFixed(2));
+
+  // Invoke adjustClasslist()
+  adjustClasslist();
+};
+
+/**
+ * Sample implementation of a currency stepper.
+ */
+
+export const Samples = {
+  parameters: {
+    controls: {
+      exclude: [
+        'title',
+        'type',
+        'size',
+        'inputmode',
+        'value',
+        'placeholder',
+        'label',
+        'help-text',
+        'message',
+        'clearable',
+        'disabled',
+        'readonly',
+        'password-toggle',
+        'password-visible',
+        'no-spin-buttons',
+        'minlength',
+        'maxlength',
+        'min',
+        'max',
+        'form',
+        'name',
+        'required',
+        'pattern',
+        'step'
+      ]
```

Comment:
In think you could whitelist via include. :)

Reply:
didn't realize that was an option